File: 110 Balanced Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param {TreeNode} root
    # @return {boolean}
    def isBalanced(self, root):
        return self.height(root)>=0
    def height(self,root):
        if root==None:
            return 0
        lh = self.height(root.left)
        rh = self.height(root.right)
        return max(lh,rh)+1 if lh>=0 and rh>=0 and abs(lh-rh)<=1 else -1

File: 110 Balanced Binary Tree/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isBalanced(TreeNode* root) {
        return height(root)>=0;
    }
    int height(TreeNode* root){
        if (root==NULL)
            return 0;
        int lh = height(root->left);
        int rh = height(root->right);
        return (lh>=0&&rh>=0&&abs(lh-rh)<=1)?max(lh,rh)+1:-1;
    }
};